pkg/node/manager/node_manager.go

```diff
@@ -121,8 +125,14 @@ func (n *NodeManager) UpdateInfo(ctx context.Context, request requests.UpdateInf
 	}
 
 	// TODO: Add a Put endpoint that takes the revision into account?
-	if err := n.nodeInfo.Add(ctx, request.Info); err != nil {
-		return nil, errors.Wrap(err, "failed to save nodeinfo during node registration")
+	if err := n.store.Add(ctx, models.NodeState{
+		Info: request.Info,
+		// the nodes approval state is assumed to be approved here, but re-use existing state
+		Approval: existing.Approval,
+		// TODO can we assume the node is connected here?
```

Comment:
Probably. It might be best to inject a heartbeat as the liveness is held in memory rather than the store and we probably want to avoid situations where the node is not in the liveness map and we accidentally send Connected rather than the default of Disconnected.